From af1a75d2c992411b947d1c0ebc2a1d7f1454ed04 Mon Sep 17 00:00:00 2001
From: Godbach <nylzhaowei@gmail.com>
Date: Wed, 2 Oct 2013 17:10:11 +0800
Subject: [PATCH 76/85] BUG/MINOR: deinit: free server map which is allocated
 in init_server_map()

Both static-rr and hash with type map-based call init_server_map() to allocate
server map, so the server map should be freed while doing cleanup if one of
the above load balance algorithms is used.

Signed-off-by: Godbach <nylzhaowei@gmail.com>
[wt: removed the unneeded "if" before the free]
---
 src/haproxy.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/haproxy.c b/src/haproxy.c
index ec9f513..bc03a73 100644
--- a/src/haproxy.c
+++ b/src/haproxy.c
@@ -975,6 +975,7 @@ void deinit(void)
 		free(p->conf.lfs_file);
 		free(p->conf.uniqueid_format_string);
 		free(p->conf.uif_file);
+		free(p->lbprm.map.srv);
 
 		for (i = 0; i < HTTP_ERR_SIZE; i++)
 			chunk_destroy(&p->errmsg[i]);
-- 
1.8.1.5

